scapy/fields.py

```diff
@@ -587,6 +592,62 @@ def fld(self):
         return self._find_fld()
 
 
+class MultiplePacketField(MultipleTypeField):
```

Comment:
can you detail what this is for?

Reply:
This field allows having multiple different packet classes based on conditions. Pretty much like `bind_layers(...)`.
I need a way to dynamically modify the "sub-"protocol of UDS. Basically the first few bytes in a UDS Packet are defined by the standard, but "complex" payloads are always OEM dependent. This means, I need special packets for payloads. In the past, I abused the `bind_layers()` function for this and therefore created all the fake-layers. With this Field, I can dynamically modify the packet class of this field based on certain conditions of the UDS packet.


Here is an example:
```
class Payload3(Packet):
    fields_desc = [
        IPField("ip", 0),
        IPField("mask", 1)
    ]

UdsRdbiPr.bind_data_record(Payload3, 0x3)

pkt = UdsRdbiPr(dataRecord=Payload3(ip="127.0.0.1")
print(repr(pkt))
>>> <UdsRdbiPr  dataIdentifier=0x3 dataRecord=<Payload3  ip=127.0.0.1 |> |>
```

Reply:
So I understand what you're trying to do, and this hasn't really been the "Scapy-way" of doing things. Take 802.11 for instance, it's a bunch of `Dot11` layers following each other but it "should" be just one packet.

I don't really like changing the paradigm. I understand what you're doing, but I would think it's confusing to have 2 very different ways of doing one thing within the same tool.

What do you think?

Reply:
I agree, overall it should be consistent. I didn't know that `Dot11` also has "fake" layers.

I'm ok with the layer as it is, I just thought it's not Scapy like to have this many fake layers in UDS (and other automotive protocols). So I experimented a bit and found this way.
What I do with this field is basically
```
ConditionalField(PacketField(), x=1)
ConditionalField(PacketField(), x=2)
```

Reply:
So if you think there is no need for a `MultiplePacketField` in Scapy, I will close this PR.

Reply:
Typically we already have two ways of doing that:
- A `PacketField` where you pass a function
- A `PacketField` with `dispatch_hook`

I do agree that those aren't obvious, but I don't believe that we need a third way.. so yeah